subiquity/models/mirror.py

```diff
@@ -32,12 +36,111 @@
 
 log = logging.getLogger('subiquity.models.mirror')
 
+DEFAULT_SUPPORTED_ARCHES_URI = "http://archive.ubuntu.com/ubuntu"
+DEFAULT_PORTS_ARCHES_URI = "http://ports.ubuntu.com/ubuntu-ports"
+
+LEGACY_DEFAULT_PRIMARY_SECTION = [
+    {
+        "arches": PRIMARY_ARCHES,
+        "uri": DEFAULT_SUPPORTED_ARCHES_URI,
+    }, {
+        "arches": ["default"],
+        "uri": DEFAULT_PORTS_ARCHES_URI,
+    },
+]
 
 DEFAULT = {
     "preserve_sources_list": False,
 }
 
 
+@attr.s(auto_attribs=True)
+class PrimaryElement(abc.ABC):
+    parent: "MirrorModel" = attr.ib(kw_only=True)
```

Comment:
I think this class needs a docstring to explain what an instance represents because after a few seconds of staring I certainly don't know :-) I think you need some kind of glossary for terms like "primary", "element", "legacy", "staged", "elected". It's not that hard for me to guess at what is meant by each word but it is a bit opaque on first reading and I might be getting something wrong.

Reply:
Updated, let me know if this is clearer now :)

I made some adjustments too, hopefully this is more sensisble:
PrimaryElement => BasePrimaryEntry
LegacyPrimarySection => LegacyPrimaryEntry
PrimaryEntry => PrimaryEntry (stays the same)